### ISSUE TITLE
Pytest warnings & pytest deprecation of direct File/Node creation now is an error

### ISSUE THREAD

**Issue by andy-maier** [pywbem/pywbem#2430]
Pytest currently issues the following warnings:

```
tests/unittest/pywbem/test_mof_compiler.py:2474
  /Users/maiera/PycharmProjects/pywbem/pywbem/tests/unittest/pywbem/test_mof_compiler.py:2474:
  PytestCollectionWarning: cannot collect test class 'TestFileErrors' because it has a __init__ constructor (from: tests/unittest/pywbem/test_mof_compiler.py)
    class TestFileErrors(object):

tests/functiontest/conftest.py:221
  /Users/maiera/PycharmProjects/pywbem/pywbem/tests/functiontest/conftest.py:221:
  PytestDeprecationWarning: direct construction of YamlFile has been deprecated, please use YamlFile.from_parent
    return YamlFile(path, parent)

tests/functiontest/conftest.py:249
  /Users/maiera/PycharmProjects/pywbem/pywbem/tests/functiontest/conftest.py:249:
  PytestDeprecationWarning: direct construction of YamlItem has been deprecated, please use YamlItem.from_parent
    yield YamlItem(tc_name, self, testcase, filepath)
```

**Comment by andy-maier** [comment]
The PytestCollectionWarning in test_mof_compiler.py is fixed in PR #xxx.

For the pytest deprecation warnings, I have not yet taken steps to address them, but have opened https://github.com/pytest-dev/pytest/issues/7696. Waiting for responses.

**Comment by andy-maier** [comment]
Since pytest 6.1.0 removed the direct creation of File/Node objects, PR #2444 now needs to be rolled back into pywbem 1.0 and 0.17.

**Comment by andy-maier** [comment]
PR #2471 rolls the fix backinto 0.17.6.
